spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java

```diff
@@ -382,4 +388,16 @@ protected void addIgnoredHeaders(String... names) {
 	boolean isSslHostnameValidationEnabled() {
 		return this.sslHostnameValidationEnabled;
 	}
+
+	// Get the header value as a long in order to more correctly proxy very large requests
+	protected long getContentLength(HttpServletRequest request) {
+		if(useServlet31){
+			return request.getContentLengthLong();
+		}
+		String contentLengthHeader = request.getHeader(HttpHeaders.CONTENT_LENGTH);
```

Comment:
Why add this change here? This wasn't part of the original issue was it?

Reply:
Well, this row would avoid having issues even for older servlets, once the header has an accurate value on it. Anyway, we could remove it and simply have getContentLength as a fallback. Please let me know what you think would be better and I will change accordingly.

Reply:
What cases specifically? Spring 5 has a servlet 3.0 minimum. Is there a test case for this?

Reply:
Not sure how we could be running tests depending on Servlet 3.0, but I could be adding tests to cover this row. The idea of trying to get that Long value from header reproduces the same fix servlet 3.1 did. [See Coyote Request](http://grepcode.com/file/repo1.maven.org/maven2/org.apache.tomcat.embed/tomcat-embed-core/8.0.24/org/apache/coyote/Request.java#Request.getContentLengthLong%28%29). 
But, again, it's up to you. If you think it's better to remove it, I can do that.